### PR TITLE
feat(profiling): span tree

### PR DESCRIPTION
**Pull request by JonasBa**
Create a first and most primitive version of the span tree (the idea is that we can have multiple ways in this class to generate a tree and we can experiment with data visualizations).

Right now, discard the notion of parent_span_id edges, sort spans by start time and search for a parent in the tree for each insertion. We track spans we fail to insert (because we sort by start time, it means only ends can overlap). Next up is to create a basic renderer to render the tree so we can see how the strategy works.

I suspect a lot of this will change with time, but the idea is to get the boilerplates in and iterate/find best strategy there.

### CODE REVIEW

File: static/app/components/events/interfaces/performance/utils.tsx [hunk 1]
```diff
@@ -96,10 +97,13 @@ export function getSpanInfoFromTransactionEvent(
   }
 
   // Let's dive into the event to pick off the span evidence data by using the IDs we know
-  const spanEntry = event.entries.find((entry: SpanEntry | any): entry is SpanEntry => {
+  const spanEntry = event.entries.find((entry: EntrySpans | any): entry is EntrySpans => {
     return entry.type === EntryType.SPANS;
   });
-  const spans: Array<RawSpanType | TraceContextSpanProxy> = [...spanEntry?.data] ?? [];
```
Comment (JonasBa) [inline static/app/components/events/interfaces/performance/utils.tsx:102]: coalescing wasnt doing anything here

File: static/app/components/events/interfaces/spans/spanTreeModel.spec.tsx [hunk 2]
```diff
@@ -85,7 +85,7 @@ describe('SpanTreeModel', () => {
         type: EntryType.SPANS,
       },
     ],
-  } as EventTransaction;
+  } as unknown as EventTransaction;
```
Comment (JonasBa) [inline static/app/components/events/interfaces/spans/spanTreeModel.spec.tsx:88]: These started failing as the type of entries was no longer data:any

File: static/app/components/events/interfaces/spans/spanTreeModel.spec.tsx [hunk 3]
```diff
@@ -550,6 +550,10 @@ describe('SpanTreeModel', () => {
       },
     };
 
+    if (!Array.isArray(event2.entries[0].data)) {
```
Comment (JonasBa) [inline static/app/components/events/interfaces/spans/spanTreeModel.spec.tsx:553]: request transaction entry is not an array so we need to guard from push on non array value (doing so narrows the type to spans[])

File: static/app/components/events/interfaces/spans/types.tsx [hunk 4]
```diff
@@ -128,11 +128,6 @@ export type EnhancedProcessedSpanType =
       type: 'span_group_siblings';
     } & SpanSiblingGroupProps);
 
-export type SpanEntry = {
```
Comment (JonasBa) [inline static/app/components/events/interfaces/spans/types.tsx:131]: This was duplicated type and I moved it to the event.tsx where other entries are defined

**Review by eliashussary — APPROVED** [review]
Looks good!

**Pull request merged**